### PR TITLE
[feat/aut835] Develop release pipeline

### PR DESCRIPTION
**Pull request by ChristopherAbram**
+ Add release workflow with prepare_release, publish_release and trigger_release jobs,
+ Rename build workflow to ci workflow.

Note:
- e-mail notifications are yet to be included (slack notifications are ready but not included in this PR)
- #324 has to be merged first.

I would like your insight, especially @ahmedskhalil

### CODE REVIEW

**Review by ghost — APPROVED** [review]
Very nice work, can't wait to see this in action.

File: .circleci/config.yml [hunk 1]
```diff
@@ -258,12 +292,211 @@ jobs:
             cp ~/SmartPeak/smartpeak_release_build/*.exe ~/artifacts
       - store_artifacts:
           path: ~/artifacts
+
+  prepare_release:
+    # This job is responsible for updating a package version stored in a configuration.
+    # The update is done autmatically and changes are pushed back to release branch.
```
Comment (dmccloskey) [inline .circleci/config.yml]:
```suggestion
    # The update is done automatically and changes are pushed back to release branch.
```

**Review by dmccloskey — APPROVED** [review]
Looks great to me as well.

**Review by bertrandboudaud — APPROVED** [review]
Great job :)

**Comment by ghost** [comment]
Congratulations :tada:. DeepCode [analyzed](https://www.deepcode.ai/app/gh/AutoFlowResearch/SmartPeak/adc835c5bd3ade61daaefa564c4a5113fd23124a/AutoFlowResearch/SmartPeak/4682a0f4e1c4cd5031a9e0104732df8a5278fa39/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=0&) your code in 1.205 seconds and we found no issues. Enjoy a moment of no bugs :sunny:.

#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/AutoFlowResearch/SmartPeak/adc835c5bd3ade61daaefa564c4a5113fd23124a/AutoFlowResearch/SmartPeak/4682a0f4e1c4cd5031a9e0104732df8a5278fa39/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=0&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=AutoFlowResearch)

**Comment by ChristopherAbram** [comment]
Retry DeepCode

**Pull request merged**